src/classes/Test_DMLManager.cls

```diff
@@ -26,12 +26,75 @@
 @isTest
 private class Test_DMLManager {
 	
+	@TestSetup
+	static void testSetup() {
+		createRestrictedUser();
+	}
+
+	/**
+	 * Creates a User that has only "Read" access to Accounts, Contacts and Opps
+	 */
+	private static void createRestrictedUser() {
+		// PMGR-9655 - Starting with Spring '21, the "Read Only" profile may not exist. We should default to using the
+		// new "Minimum Access - Salesforce" profile, falling back to "Read Only" otherwise.
+		List<Profile> profileList = [SELECT Id, Name FROM Profile WHERE Name IN ('Minimum Access - Salesforce', 'Read Only') ORDER BY Name];
+		System.assert(
+			profileList.size() >= 1,
+			'Unable to create a "restricted" User for testing purposes because we are not able to find the required User Profiles.'
+		);
+		User restrictedUser = new User(
+			Alias = 'standt',
+			Email='standarduser@testorg.com',
+			EmailEncodingKey='UTF-8',
+			LastName='Testing',
+			LanguageLocaleKey='en_US',
+			LocaleSidKey='en_US',
+			// Use the first Profile in the list. Because we've ordered by name, this causes us to use
+			// "Minimum Access - Salesforce" if it's present.
+			ProfileId = profileList[0].Id,
+			TimeZoneSidKey='America/Los_Angeles',
+			Username='crudmanageruser1@testorg.com'
```

Comment:
Should we use an Alias to identify our user vs. this Username that might result in transient build failures if multiple SF orgs are running tests at the same time?

Reply:
Technically, this is not new code. This was taken from the old ensureRestrictedUser method at the bottom of this test class, so, it has been like this for a long time. I'm not saying we shouldn't consider changing this to generate unique usernames, I'm just saying that I think the risk of username collisions is low, otherwise we would have seen evidence by now. I'm more concerned about using "alias". Alias is only 8 characters long, and is not required to be unique. I think that makes it much more likely to collide with other Users with the same Alias created during parallel test executions.

Reply:
Point taken; you can merge as-is